cvat/apps/dataset_manager/views.py

```diff
@@ -127,14 +176,55 @@ def export_project_annotations(project_id, dst_format=None, server_url=None):
     return export(dst_format, project_id=project_id, server_url=server_url, save_images=False)
 
 
-def clear_export_cache(file_path, file_ctime, logger):
+def clear_export_cache(file_path: str, file_ctime: float, logger: logging.Logger) -> None:
     try:
-        if osp.exists(file_path) and osp.getctime(file_path) == file_ctime:
+        with get_dataset_cache_lock(
+            file_path,
+            block=True,
+            acquire_timeout=EXPORT_CACHE_LOCK_TIMEOUT,
+            ttl=rq.get_current_job().timeout,
+        ):
+            if 'job' in file_path:
```

Comment:
This seems even less robust than `get_file_instance_timestamp`. `job`, `task`, `project` are all common words, and I could easily see them occurring in the file path somewhere other than expected.

Reply:
Improved the parsing a little.